### ISSUE TITLE
include photo source and original url

### ISSUE THREAD

**Issue by andrewharvey** [inaturalist/iNaturalistAPI#188]
`GET /observations/{id}` will return a list of photos in the form

```json
 {
          "id": 61482854,
          "license_code": "cc-by",
          "url": "https://static.inaturalist.org/photos/61482854/square.jpg?1581761020",
          "attribution": "(c) Andrew Harvey, some rights reserved (CC BY)",
          "original_dimensions": {
            "width": 2048,
            "height": 1536
          },
          "flags": []
        }
```

and a list of observation_photos in the form:

```json
{
          "id": 56964180,
          "position": 0,
          "uuid": "454c6b2d-8d4d-4b6c-81ed-492f74d310f6",
          "photo": {
            "id": 61482854,
            "license_code": "cc-by",
            "url": "https://static.inaturalist.org/photos/61482854/square.jpg?1581761020",
            "attribution": "(c) Andrew Harvey, some rights reserved (CC BY)",
            "original_dimensions": {
              "width": 2048,
              "height": 1536
            },
            "flags": []
          }
        }
```

However it doesn't include any other metadata like the original source eg https://www.inaturalist.org/photos/61482854 links to the Flickr ID.

Can either the source + original URL be included in the response or could there be a separate GET /photo/{id} request to provide full photo metadata?

**Comment by kueda** [comment]
Sorry, somehow this one slipped by me. Original URLs are always at `https://static.inaturalist.org/photos/${photo_id}/square.${ext}` so they can be derived from the `url` attribute you get in the observation responses. We don't really have an internal need for a separate photo detail endpoint, so we're probably not going to build one ourselves. What's the use case you had in mind?

**Comment by andrewharvey** [comment]
hi @kueda for example my observation at https://www.inaturalist.org/observations/38664139 has 5 photos for example the first one is https://www.inaturalist.org/photos/61482854. The response at https://api.inaturalist.org/v1/observations/38664139 provides the metadata for these photos eg.

```json
{
  "id": 61482854,
  "license_code": "cc-by",
  "url": "https://static.inaturalist.org/photos/61482854/square.jpg?1581761020",
  "attribution": "(c) Andrew Harvey, some rights reserved (CC BY)",
  "original_dimensions": {
    "width": 2048,
    "height": 1536
  },
  "flags": []
},
```

However if I inspect the photo metadata at https://www.inaturalist.org/photos/61482854 it reveals a lot more information like the photo was imported from Flickr and the original Flickr URL was https://www.flickr.com/photos/136319147@N08/49537447541/

That's the part I can't find an API response for. Photos has a POST operation but not GET and it's not included in the observation api response, so the only way I can see is to scrape the HTML page at https://www.inaturalist.org/photos/61482854 which isn't ideal.

You probably won't like my use case, it's for working around observation coordinates that are hidden by iNat but public at Flickr :wink: 

Secondary use case is to provide attribution back to the original Flickr source, not just to the iNat photo page.

At the same time I've been investigating using the gbif data dump instead, but it would still be nice for the API to provide this.

**Comment by kueda** [comment]
> You probably won't like my use case, it's for working around observation coordinates that are hidden by iNat but public at Flickr 

You're right, I don't like that and am not going to implement anything just to support it.

> Secondary use case is to provide attribution back to the original Flickr source, not just to the iNat photo page

I'd be ok with that, but frankly if you got the image from iNat, why not just cite the iNat page? The instance on Flickr might not be the "original" either. IMO, best citation practice for images on the web is abide by the license and link to the place you found it with the license declaration you're abiding by.

Anyway, thanks for replying and explaining. I'm not opposed to a photo detail endpoint in principle, or including the sourcing information we have in our database, so if someone else wants to build that, cool. I'll just leave this open. FWIW, I don't think this is worth making a separate photos elasticsearch index (though I can think of other good uses for that).

**Comment by andrewharvey** [comment]
I just discovered that the "old" API at https://www.inaturalist.org/pages/api+reference#get-observations-username does return Photo origins (eg Flickr) and the original Flickr URL, so as a workaround I'll use the old API.

Happy to leave this ticket open for equivalent support in the new API.

**Comment by andrewharvey** [comment]
I wrote about my use case of linking my Flickr photos with iNaturalist observations at https://tech.beyondtracks.com/posts/inaturalist-integration/, it was only possible to do because of the old API support, so still very keen to see the new API support this.

**Comment by npdoty** [comment]
> Original URLs are always at `https://static.inaturalist.org/photos/${photo_id}/square.${ext}` so they can be derived from the `url` attribute you get in the observation responses. We don't really have an internal need for a separate photo detail endpoint, so we're probably not going to build one ourselves. What's the use case you had in mind?

Could we document this URL structure in the new API documentation? I was looking for a photo detail endpoint, but just because I wanted to find the URL for the medium-size or original-size photo rather than the square icon. But now it seems like maybe I can accomplish that just by changing `square.ext` to `medium.ext`. If that's just generally true, that helps a lot, but not everyone may find this github issue. Also, what is the enumeration of size names?

**Comment by kueda** [comment]
@npdoty that seems like a separate issue, so I made https://github.com/inaturalist/iNaturalistAPI/issues/299
